Unity-Technologies/ScriptableRenderPipeline
# Support for Camera Relative Position Shader Graph + HDRP

### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?
This allows Shader Graph to properly and explicitly support camera relative world space position when using the HDRP in a project. **World** space now uses the default world space coordinates of the active render pipeline, and **Absolute World** provides the absolute world space coordinates regardless of active pipeline.
![image](https://user-images.githubusercontent.com/39529353/59060879-f88bfe80-8856-11e9-8cb7-eaca744653b2.png)


---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.
Previous graphs made using the Position node in World space when upgrading to this version will be automatically set to Absolute World, to make sure that calculations maintain the expected output. 

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: What did you do?
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)
**Input** test has been updated to include Absolute World Position. 
![InputNodes](https://user-images.githubusercontent.com/39529353/59061109-7cde8180-8857-11e9-90cf-55cb43358478.png)

**Transforms** test has been added to confirm all position transforms from one coordinate space to another function properly, shown below with output for World and Absolute World in LWRP (currently the same output).
![TransformNode](https://user-images.githubusercontent.com/39529353/59061112-7fd97200-8857-11e9-9628-f8ea0418ea01.png)


Launch Katana (Link on master here - update for your branch):
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=sg%2Fhdrp-position&unity_branch=trunk&automation-tools_branch=master&sort=1-asc

Alternative, launch Yamato (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/sg%252Fhdrp-position

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High? Medium, changes shader generation code for both LWRP and HDRP.

**Halo Effect**: None, Low, Medium, High? Medium, changes locations of helper functions and adds serialized data to the graph 

---
### Comments to reviewers
Notes for the reviewers you have assigned.
